spring-boot-project/spring-boot-test/src/main/java/org/springframework/boot/test/context/FilteredClassLoader.java

```diff
@@ -70,6 +82,16 @@ public FilteredClassLoader(Predicate<String>... filters) {
 		return super.loadClass(name, resolve);
 	}
 
+	@Override
+	public URL getResource(String name) {
+		for (Predicate<String> filter : this.filters) {
```

Comment:
This may be overkill but classes and resources predicates are being applied to both use cases with no way for an implementation to know what type of "resource" needs to be filtered.

Reply:
No exactly, this is why I found the current design a bit unfortunate. If there's no objection to changing FilteredClassLoader a bit more (and adjusting the tests currently using it), I could take a stab at refactoring into a builder idea, similar to `ApplicationContextRunner`.